src/main/scala/uk/co/gresearch/spark/diff/Diff.scala

```diff
@@ -150,15 +195,11 @@ object Diff {
     "_" * (existing.map(_.length).reduceOption(_ max _).getOrElse(0) + 1)
   }
 
-  def of[T](left: Dataset[T], right: Dataset[T], idColumns: String*): DataFrame =
-    default.of(left, right, idColumns: _*)
+  def of[T](left: Dataset[T], right: Dataset[T], idColumns: String*)(implicit diff: Diff = default): DataFrame =
```

Comment:
Definitively an improvement of the API, should be done in a separate PR and in a non-breaking way. I think I though of this initially but did not do it because of some ambiguity.